lib/services/assessment.js

```diff
@@ -334,32 +334,78 @@ module.exports = class AssessmentService extends Schmervice.Service {
     async recordStageTranisiton(student, toStage, txn=null){
         const { StageTransition } = this.server.models();
         const { exotelService } = this.server.services();
-
         await StageTransition.query(txn).insert({
             studentId:student.id,
             fromStage:student.stage,
             toStage,
         });
 
-        // send sms after recording the stage transitions
-        if ( exotelService.hasTemplateForStage(toStage) == true ) {
-            if (!student.contacts) {
-                await student.$relatedQuery('contacts');
-            }
-            let sendSMSPromises = []
-            _.each(student.contacts, contact => {
-                const templateContext = {
-                    student,
-                    contact
+        // send sms after recording the stage transitions if stage is not related to the online/offline test result. 
+        if (toStage != "testPassed" && toStage != 'testFailed'){
+            if ( exotelService.hasTemplateForStage(toStage) == true ) {
+                if (!student.contacts) {
+                    await student.$relatedQuery('contacts');
                 }
-                sendSMSPromises.push(exotelService.sendSMS(contact.mobile, toStage, templateContext))
-            })
-            return await Promise.all(sendSMSPromises)
+                let sendSMSPromises = []
+                _.each(student.contacts, contact => {
+                    const templateContext = {
+                     student,
+                        contact
+                    }
+                    sendSMSPromises.push(exotelService.sendSMS(contact.mobile, toStage, templateContext))
+                })
+                return await Promise.all(sendSMSPromises)
+            }
         }
         
     }
+    
+    // Inform test result after 4 hours using CRON.
+    async informTestResult() {
```

Comment:
The point of this function should be to just update the stage transition using patchStudent. That method should automatically ensure that the SMS is being sent. It shouldn't be the work of this code.